crates/cast/src/main.rs

```diff
@@ -372,3 +407,59 @@ fn update_cast_config(config: &mut CastConfig, cli: &Cli) {
 
     config.accounts_file = Utf8PathBuf::from(shellexpand::tilde(&new_accounts_file).to_string());
 }
+
+struct PackageData {
+    pub manifest_path: Result<Utf8PathBuf>,
+    pub metadata: Result<Metadata>,
+    pub package: Result<PackageMetadata>,
+}
+
+// Lazily get all the required package information from CLI
+fn get_package_data(cli: &Cli) -> PackageData {
+    let mut result = PackageData {
+        manifest_path: Err(anyhow!("Could not retrieve the manifest path")),
+        metadata: Err(anyhow!("Could not retrieve metadata")),
+        package: Err(anyhow!("Could not retrieve package metadata")),
```

Comment:
The "no Scarb" error didn't change. The code uses those default messages only when the required previous field wasn't found (eg. there is no metadata when there's no manifest path) in practice these should barely (or never) show up

Reply:
So I tried to check this, and it seems the behaviour is as you described when we are in the directory with Scarb.toml. However this changes when we pass the path to it:

```
# happy case
$ which scarb
/Users/wojciech/.asdf/shims/scarb

$ cargo run -- --path-to-scarb-toml scripts/Scarb.toml --url http:/127.0.0.1:5055/rpc --account user1 script map_script
command: script
status: success

# remove scarb from PATH
$ export PATH= (...)
$ which scarb
scarb not found

$ cargo run -- --path-to-scarb-toml scripts/Scarb.toml --url http:/127.0.0.1:5055/rpc --account user1 script map_script
Error: Could not retrieve package metadata

# now go into the dir
$ cd scripts
$ cargo run -- --url http:/127.0.0.1:5055/rpc --account user1 script map_script
Error: Failed to obtain manifest path from scarb

Caused by:
    0: Cannot find `scarb` binary in PATH. Make sure you have Scarb installed https://github.com/software-mansion/scarb
    1: cannot find binary path
```

so that's a bit odd

Reply:
I've moved the scarb check higher - we require it anyway and it looks like it was the exception to this rule.